### ISSUE TITLE
Can the date picker be shown as a gregorian calendar month?

### ISSUE THREAD

**Issue by vvizden** [element-plus/element-plus#827]
Hello,

the date picker for month looks like a lunar calendar, it looks very confusing. Can that be shown as a gregorian calendar month?

The locale lang is zh-cn.

**Comment by zazzaz** [comment]
Any detail or screenshot, pls?

**Comment by JeremyWuuuuu** [comment]
@vvizden Any updates?

**Comment by vvizden** [comment]
Maybe I didn't make myself clear. The month picker just looks like a lunar calendar, such as 十二月. I think 12月 may be probably better, it's more like the Gregorian calendar.

**Comment by JeremyWuuuuu** [comment]
> > @vvizden Any updates?
> 
> Maybe I didn't make myself clear. The month picker just looks like a lunar calendar, such as 十二月. I think 12月 may be probably better, it's more like the Gregorian calendar.

Well, I think I get it, this is something related to locale, because that was translated from `locale` file, In `dayjs` I believe that there is a work around on this. cc @zazzaz

**Comment by vvizden** [comment]
> > > @vvizden Any updates?
> > 
> > 
> > Maybe I didn't make myself clear. The month picker just looks like a lunar calendar, such as 十二月. I think 12月 may be probably better, it's more like the Gregorian calendar.
> 
> Well, I think I get it, this is something related to locale, because that was translated from `locale` file, in `dayjs` I believe that there is a work around on this. cc @zazzaz

It is better to show with the Gregorian calendar, because it is different from the lunar calendar.

**Comment by zazzaz** [comment]
It's not about Gregorian or lunar calendar.

In month picker, we list months via month name, that's Jan, Feb...

And the two Chinese translation '1月' and '一月' are both correct.

**Comment by vvizden** [comment]
> It's not about Gregorian or lunar calendar.
> 
> In month picker, we list months via month name, that's Jan, Feb...
> 
> And the two Chinese translation '1月' and '一月' are both correct.

I know that both of these translation are correct, it just looks more like the lunar calendar at now. on the other way, We are ususlly used to using a similar format such as `2020年12月` or `二〇二〇年十二月`, but not `2020年十二月`. Consider the consistency of them?

**Comment by JeremyWuuuuu** [comment]
@vvizden , what exact problem you try to ask us to solve here? I think Chineses all know `12月` and `十二月` instead of `农历十二月`, if this is not for Chinese users to view then why are we using Chinese as the locale at all? I am kind of confused.

**Comment by JeremyWuuuuu** [comment]
I don't think that you are wrong but I personally think that everybody that can read interpret `十二月` as December in Gregorian calendar. When it's like `腊月` we say that's Lunar calendar. Right?

**Comment by vvizden** [comment]
> @vvizden , what exact problem you try to ask us to solve here? I think Chineses all know `12月` and `十二月` instead of `农历十二月`, if this is not for Chinese users to vie

> I don't think that you are wrong but I personally think that everybody that can read interpret `十二月` as December in Gregorian calendar. When it's like `腊月` we say that's Lunar calendar. Right?

Yes, you are right. But `2020年12月` is better than `2012年十二月`, that's right?

**Comment by vvizden** [comment]
> I don't think that you are wrong but I personally think that everybody that can read interpret `十二月` as December in Gregorian calendar. When it's like `腊月` we say that's Lunar calendar. Right?

Excluding `正月`, `冬月` and `腊月`, for the others, older people may feel more like the lunar calendar at first glance.

**Comment by vvizden** [comment]
If you don't think so, I can close this issue.

**Comment by JeremyWuuuuu** [comment]
> If you don't think so, I can close this issue.

I don't think this is something with exact right or wrong, it's something that we need to discuss about, especially for the users who actually uses pages built upon this library.

I have already seen your point here. It's for the elders to understand that the calendars they saw was not lunar calendar.

**Comment by JeremyWuuuuu** [comment]
Maybe we can discuss some solution for customization here.

**Comment by vvizden** [comment]
If there is a functional requirement, we need to switch between lunar and Gregorian calendars. How should the month picker be listed? `十二月` vs `腊月` or `12月` vs `腊月`, also `十月` vs `十月` or `10月` vs `十月`?

**Comment by JeremyWuuuuu** [comment]
I think when its lunar calendar, `腊月` seems better to me. But for me to distinguish the Lunar calendar and the Gregorian calendar, it depends on whether they have Chinese character like for today `廿四` underneath the numbers

**Comment by vvizden** [comment]
> I think when its lunar calendar, `腊月` seems better to me. But for me to distinguish the Lunar calendar and the Gregorian calendar, it depends on whether they have Chinese character like for today `廿四` underneath the numbers

Maybe it's just a matter of personal habit. That's just my sudden thought  someday, why not using this format, such as `1月`, `2月` ... ? Well, that's all.

**Comment by JeremyWuuuuu** [comment]
Well, changing this is easy based on what @zazzaz told me, just a couple of strings, I'll leave the rest discussion to him, this won't take long, I suppose, besides, `AntDesign` did as you proposed `1月` instead of `一月`.

**Comment by zazzaz** [comment]
By design, element plus will use date-time-related locale files from Day.js directly.

As for the abbreviated month name in this issue we've talked about, Day.js shows:

https://github.com/iamkun/dayjs/blob/dev/src/locale/zh-cn.js#L10
`monthsShort: '1月_2月_3月_4月_5月_6月_7月_8月_9月_10月_11月_12月'.split('_'),`

That is to say, month picker will finally be the exact result you are looking for.

**Comment by vvizden** [comment]
> By design, element plus will use date-time-related locale files from Day.js directly.
> 
> As for the abbreviated month name in this issue we've talked about, Day.js shows:
> 
> https://github.com/iamkun/dayjs/blob/dev/src/locale/zh-cn.js#L10
> `monthsShort: '1月_2月_3月_4月_5月_6月_7月_8月_9月_10月_11月_12月'.split('_'),`
> 
> That is to say, month picker will finally be the exact result you are looking for.

Hi, i traced ths source code of element+, i found that element+ didn't use the locale files from Day.js, but the files in `element-plus/lib/locale/lang/`.

Default `lang` is `element-plus/lib/locale/lang/en`, of course we can register a new locale file for 'zn-cn' instead of `element-plus/lib/locale/lang/zh-cn.ts` when using element+.

Do you have any plans to modify the locale file at `element-plus/lib/locale/lang/zh-cn.ts` ?

**Comment by zazzaz** [comment]
will do to replace to dayjs/locale/monthsShort

**Comment by vvizden** [comment]
> will do to replace to dayjs/locale/monthsShort

Cool ! However,  here is a light digression. Althought Day.js is small enough, are there any plans for Element+ to use libraries similar to `date-fns` for tree shaking?

**Comment by zazzaz** [comment]
why `date-fns`? even with tree shaking, `date-fns` is still 10x larger than `Day.js`

**Comment by vvizden** [comment]
> why `date-fns`? even with tree shaking, `date-fns` is still 10x larger than `Day.js`

Awesome!
